src/core/highlight.js

```diff
@@ -58,7 +58,12 @@ export async function run(conf) {
         if (id !== msg.id) {
           return; // not for us!
         }
-        element.innerHTML = value;
+        const codeElements = element.querySelectorAll('code');
+        if(codeElements.length > 0)
+          element.innerHTML = value;
+        else {
+          element.innerHTML = `<code>${value}</code>`;
+        }
         if (element.localName === "pre") {
           element.classList.add("hljs");
```

Comment:
Looks okay if we only want simple wrapping, but I think new classes should also go `<code>` instead of `<pre>`. Not exactly sure about `hljs` but at least the language hint should. Thoughts? @marcoscaceres

Reply:
new classes as in ?

Reply:
I mean:

```html
<pre class="js">
...code...
</pre>
```

should become:

```html
<pre><code class="js">
...code...
</code></pre>

Reply:
(Note that we have `utils.renameElement()`, so we may just "rename" the `<pre>` into a `<code>` and then add it to a new `<pre>`.)

Reply:
`class="js"` is just an example right ? normally, we won't add a `language-class`  to `<pre>` bcz the highlighter identifies the language by itself ?

Reply:
> I mean:
> 
> ```
> <pre class="js">
> ...code...
> </pre>
> ```
> 
> should become:
> 
> ```
> <pre><code class="js">
> ...code...
> </code></pre>
> ```

You mean any class that is associated with `<pre>` also needs to added to `<code>` or only the language class, bcz only that'd be relevant to the `<code>` tag right ?

Reply:
We shouldn't transfer everything, but only language class and `hljs`.

My suggestion:

1. If `<pre>` have no `<code>` then `utils.wrapInner()` by `<code>` but still use the class names from `<pre>` to get language hints for highlighter.
2. The highlighter will return the exact language name later. Add it to `<code>`, and remove it from `<pre>`.

Reply:
There is a specific format for writing a language class in `<pre>` right ? `class="lang-js"` prefixed with `lang` ?

Reply:
When the `<pre>` has no language class mentioned, it predicts it being `javascript` almost every time 😛  I tried with c++/python it still returns `language: ["javascript"]` from the worker.

Reply:
The highlight bundle deliberately only knows about html/xml, css, js, http, and a very restricted set of others. It’s be design that it only matches those - c++ or phython won’t ever match because we deliberately don’t support them.

Reply:
The changes, do they look good ? It's for the case when no `<code>` element is found inside of `<pre>`